### PR TITLE
Feature add remove

### PR DESCRIPTION
**Pull request by Amalcxc**
- Remove all hardcoded items from the tasks array.

- Create a new JavaScript file for the new functionality.

- Implement a function for adding a new task (add a new element to the array).

- Implement a function for editing task descriptions.

- Implement a function for deleting a task (remove an element from the array).

- Implement a function for the "Clear all completed".

### CODE REVIEW

File: src/items.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      const newId = event.target.parentNode.getAttribute('id');
+      tasks[newId].description = newTask;
+      update(tasks);
+      const anotherT = document.querySelector(`li[id="${newId}"]`);
+      anotherT.innerHTML = newTask;
+    });
+  });
+}
+
+function removeLocalTodos(tasks) {
```
Comment (vichuge) [inline src/items.js:101]: - [x] When you remove an element, the index should be refreshed according to the list, always the first element should be one, the second two, please fix this issue. 🐶 
image

File: src/items.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    todo.remove();
+    removeLocalTodos(todo);
+  }
+}
+
+list.addEventListener('click', deleteitem);
+
+const clearAll = document.querySelector('.clear');
+
+function clear() {
```
Comment (vichuge) [inline src/items.js:126]: - [x] Here on clear is a little diff as expected, when I have some elements, for example:
image
- Then, I delete:
image
- This is the result if I refresh the page:
image
- I think this problem is about the id's issue, please, check this bug and if the id is good, maybe this issue is solved too. 🐱

**Review by vichuge — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi dear student, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 
You used proper GitHub Flow ✔️
Your linter checks are all passing ✔️
Your PR messages are descriptive ✔️

## Aspects to improve ♻️

- [x] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/items.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+function deleteitem(e) {
+  const item = e.target;
+  if (item.classList[0] === 'trash-btn') {
+    const todo = item.parentElement;
+    todo.remove();
+    removeLocalTodos(todo);
+  }
+}
```
Comment (MahmoudBakr23) [inline src/items.js:120]: - [ ] **OPTIONAL**
Great job managing the resorting issue 👏 However, it seems to be only showing the results after refreshing. I encourage you to put some more time on that 😄 🙏

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Amalcxc 😃,

Well done implementing the required changes from the previous review. However, I've got some **OPTIONAL** but great suggestions for you down below! 😉 ❤️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

**OPTIONAL**
- Please consider updating the live demo link to match the app's latest version. This is for a more professional and consistent repo. 😄 
- Also, please always keep in mind in the future to write a more descriptive PR headline (not the default branch name) This helps viewers to quickly get what this PR is actually about 🥰 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**